react-client/src/App.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from "react";
+import "./App.css";
+import DatasourceCredentialsForm from "./DatasourceCredentialsForm";
+import TextArea from "./TextArea";
+
+export default function App() {
+  const api = "http://localhost:9099";
+  const [query, setQuery] = useState("");
+
+  //TODO: refactor duplicate in DatasourceCredsForm
+  const initCreds = {
+    username: "",
+    host: "",
+    password: "",
+    dbname: "",
+  };
+
+  /* TODO: set expected response headers */
+  const handleSubmit = (formdata: any) => {
+    fetch(api + "/datasources", {
+      method: "POST",
+      body: JSON.stringify({
+        data: {
+          type: "datasource",
+          attributes: { ...formdata },
+        },
+      }),
+    })
+      .then(function (response) {
```

Comment:
use `async/await` instead of promise-chaining:
```
try {
  const response = await fetch(api + "/datasources", {
    method: "POST",
    body: JSON.stringify({
      data: {
        type: "datasource",
        attributes: { ...formdata },
      },
    }),
  })
  if (response.ok) {
    return await response.json()
  }
```
So much better!